src/components/result-catalog/ResultCatalog.tsx

```diff
@@ -0,0 +1,151 @@
+import { FC, useEffect, useState } from 'react';
+import { fetchData } from '../../service/apiService';
+import { ApiInfo, IApi } from '../../utils/types/types';
+import ResultCard from '../result-card/ResultCard';
+import Pagination from '../pagination/Pagination';
+import {
+  Link,
+  Outlet,
+  Route,
+  Routes,
+  useNavigate,
+  useParams,
+} from 'react-router-dom';
+import '../components.css';
+import SideBar from '../sidebar/SideBar';
+import SelectItemPerPage from '../select/SelectItemPerPage';
+interface ResultCatalogProps {
+  queryParam: string;
+  startPage: boolean;
+  handleStopSearch: () => void;
+}
+
+const ResultCatalog: FC<ResultCatalogProps> = (props) => {
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [items, setItems] = useState<IApi[]>([]);
+  const [apiInfo, setApiInfo] = useState<ApiInfo>();
+  const [hasError, setHasError] = useState(false);
+  const { page } = useParams();
+  const [currentPage, setcurrentPage] = useState(Number(page));
+  const [isSideBarOpen, setIsSideBarOpen] = useState(false);
+  const [totalCard, setTotalCard] = useState(20);
+  const navigate = useNavigate();
+  const defaultApiCardPerPage = 20;
+  const startApiPage =
+    (Number(page) - 1) * (totalCard / defaultApiCardPerPage) + 1;
+
+  const totalPages = Math.ceil(totalCard / defaultApiCardPerPage);
+
+  useEffect(() => {
+    async function fetchDataForAllPages(
+      queryParam: string | undefined,
+      startPage: boolean
+    ) {
+      setIsLoaded(false);
+      setHasError(false);
+      props.handleStopSearch();
+
+      const allResults = [];
+      let pageApi = startApiPage;
+      let totalApiPages = 0;
+
+      try {
+        while (true) {
+          const data = await fetchData(queryParam, pageApi, startPage);
+          allResults.push(...data.results);
+          setApiInfo(data.info);
+          totalApiPages = data.info.pages;
+
+          if (
+            pageApi < startApiPage + totalPages - 1 &&
+            pageApi < totalApiPages
+          ) {
+          } else {
+            break;
+          }
+          pageApi++;
+        }
+
+        setItems(allResults);
+        setIsLoaded(true);
+      } catch (error) {
+        console.error('Fetch error:', error);
+        setIsLoaded(true);
+        setHasError(true);
+      }
+    }
```

Comment:
This method is not ok. I'll consider how to refactor it.

Reply:
Will be irrelevant when I implement Redux